lib/nice_partials/partial.rb

```diff
@@ -94,18 +92,34 @@ def section_from(name)
       @sections ||= {} and @sections[name] ||= Section.new(@view_context, @local_assigns&.dig(name))
     end
 
+    def respond_to_missing?(meth, include_private = false)
+      meth != :to_ary # Avoid creating a section when doing `*partial`.
+    end
+
     def method_missing(meth, *arguments, **keywords, &block)
-      if @view_context.respond_to?(meth)
-        @view_context.public_send(meth, *arguments, **keywords, &block)
-      else
-        define_accessor meth and public_send(meth, *arguments, **keywords, &block)
-      end
+      define_accessor meth and public_send(meth, *arguments, **keywords, &block)
     end
 
     def define_accessor(name)
       name = name.to_s.chomp("?").to_sym
       self.class.define_method(name) { |content = nil, **options, &block| section(name, content, **options, &block) }
       self.class.define_method("#{name}?") { section?(name) }
     end
+
+    def helpers_context
+      @helpers_context ||= Helpers.new(@view_context)
+    end
+
+    class Helpers
```

Comment:
Are you fond of `SimpleDelegator`?

```diff
diff --git a/lib/nice_partials/partial.rb b/lib/nice_partials/partial.rb
index 4f1570c..e26f5a3 100644
--- a/lib/nice_partials/partial.rb
+++ b/lib/nice_partials/partial.rb
@@ -110,12 +110,7 @@ module NicePartials
       @helpers_context ||= Helpers.new(@view_context)
     end
 
-    class Helpers
-      def initialize(view_context)
-        @view_context = view_context
-      end
-      delegate_missing_to :@view_context
-
+    class Helpers < SimpleDelegator
       def self.method_added(name)
         super
         NicePartials::Partial.delegate name, to: :helpers_context
```

Reply:
Ooooo, I thought about trying something with `DelegateClass`, but I actually didn't know about `SimpleDelegator`. Nice!

Reply:
Looks like `SimpleDelegator` doesn't forward private methods, but I don't know how common it is to have those on the view side https://github.com/ruby/ruby/blob/a528908271c678360d2d8ca232c178e7cdd340b4/lib/delegate.rb#L104